staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -667,5 +668,5 @@ func compareResults(r reflect.Value, expectedVal string) (bool, error) {
 		return false, errors.New("jsonpath leads to a nested object or list which is not supported")
```

Comment:
According to the requirement, when JSONPath condition is not defined, this should return `true` because the object fits to the definition of the "_any non empty_" value.

Reply:
A nested object or list is technically a "_non-empty value_", but it is still something whose evaluation is not supported currently. I feel it would be incorrect here to return `true` in this scenario. If we are to look at it this way, won't that render the verifications done in `verifyParsedJSONPath()`  invalid? This function raises errors when the  `jsonpath` expression matches more than one list/value, which is also technically a "_non-empty value_"
Maybe this can be done once nested objects/lists and multiple values are supported to wait on

Reply:
The feature request's requirement is to allow omitting a JSON value and when the value is not provided, to wait for a JSON expression to have a non-empty value. That value should not be further evaluated or processed. Determination of the value as "non-empty" is a stopping condition. Using the original example with a service of the type `LoadBalancer`, there is no need to wait for `.status.loadBalancer.ingress[:].ip`. It will be sufficient to wait for `.status.loadBalancer`.
The unsupported part is for the use case when a caller _provides_ a value. In this case, only literals are supported.

Reply:
Got it.
Is it safe to assume that if the `jsonpath` expression evaluates to any value which is not `""`, it is a "non-empty value"?

Reply:
I believe it is. This is how the value was interpreted so far and is defined in JSON standard AFAIK